### PR TITLE
Add option to not default for C++ (which matches clang/gcc behaviour)

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #10872

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -914,13 +892,8 @@ def need_llvm_debug_info(options):
...
     options, settings_changes, newargs = parse_args(newargs)
 
-    if use_cxx:
-      clang_compiler = CXX
-    else:
-      clang_compiler = CC
-
     if '-print-search-dirs' in newargs:
-      return run_process([clang_compiler, '-print-search-dirs'], check=False).returncode
+      return run_process([CC, '-print-search-dirs'], check=False).returncode
```
Comment (kripken) [inline emcc.py:896]: shouldn't this use CXX if we are in C++ mode? (there might be other search dirs in that case?)

File: src/settings.js [hunk 2]
```diff
@@ -1783,6 +1784,11 @@ var USE_OFFSET_CONVERTER = 0;
 // [link]
 var LLD_REPORT_UNDEFINED = 0;
 
+// Default to c++ mode even when run as `emcc` rather then `emc++`.
+// When this is disabled `em++` is required when compileing and linking C++
```
Comment (kripken) [inline src/settings.js:1788]: unnecessary `e` in `compiling`

File: emcc.py (hunk 1 above)
Comment (sbc100) [reply emcc.py:896]: Apparently this doesn't differ `diff <(clang -print-search-dirs) <(clang++ -print-search-dirs)` is empty as is `diff <(gcc -print-search-dirs) <(g++ -print-search-dirs)`

Comment (sbc100) [reply emcc.py:896]: `-print-search-dirs` does not report include paths, for example, which might differ.

**Comment by sbc100** [comment]
Wrote some more extensive tests.  Also, don't add C++ include paths when not in C++ mode.

**Pull request merged**